packages/components/src/components/form/AutocompleteMultiSelect/AutocompleteMultiSelect.tsx

```diff
@@ -0,0 +1,269 @@
+        case useCombobox.stateChangeTypes.ItemClick:
+          return { ...changes, isOpen: true };
+        default:
+          return changes;
+      }
+    },
+    onStateChange({ inputValue: newInputValue, type, selectedItem: newSelectedItem }) {
+      switch (type) {
+        case useCombobox.stateChangeTypes.InputKeyDownEnter:
+        case useCombobox.stateChangeTypes.ItemClick:
+        case useCombobox.stateChangeTypes.InputBlur:
+          if (newSelectedItem) {
+            addSelectedItem(newSelectedItem);
+          }
+          setInputValue('');
+          break;
+        case useCombobox.stateChangeTypes.InputChange:
+          setInputValue(newInputValue);
+          break;
+        default:
+          break;
+      }
+    },
+    ...useComboboxProps,
+  });
+
+  return (
+    <AutocompleteMultiSelectProvider
+      value={{
+        isOpen,
+        size,
+        isDisabled,
+        items: filteredItems,
+        getMenuProps,
+        getInputProps,
+        getItemProps,
+        getToggleButtonProps,
+        selectedItem: selectedItemCombobox,
+        highlightedIndex,
+        selectedItems,
+        removeSelectedItem,
+        getSelectedItemProps,
+        getDropdownProps,
+        reset: () => setInputValue(''),
+      }}
+    >
+      <Box position="relative">{children}</Box>
+    </AutocompleteMultiSelectProvider>
+  );
+};
+
+const AutocompleteMultiSelectInput: FC<AutocompleteMultiSelectInputProps> = ({
+  clearButtonMode = 'has-value',
+  size,
+  isDisabled,
+  ...rest
+}) => {
+  const {
+    reset,
+    getInputProps,
+    getToggleButtonProps,
+    getDropdownProps,
+    isOpen,
+    size: autocompleteMultiselectSize,
+    isDisabled: autocompleteMultiSelectDisabled,
+  } = useAutocompleteMultiSelectContext();
+  const ref = useRef<HTMLInputElement>(null);
+  const inputProps = getInputProps({ ...getDropdownProps({ preventKeyAction: isOpen, ref }) });
+
+  const clearButtonConditions = {
+    'has-value': inputProps.value,
+    never: false,
+    always: true,
+  };
+  const shouldShowClearButton = clearButtonConditions[clearButtonMode];
+  const inputSize = size ?? autocompleteMultiselectSize;
+  const isInputDisabled = isDisabled || autocompleteMultiSelectDisabled;
+
+  const onReset = useCallback(() => {
+    reset?.();
+    ref?.current?.focus?.();
+  }, []);
+
+  return (
+    <Input
+      trailingIcon={
+        shouldShowClearButton ? (
+          <IconX data-testid="cmpsr.autocompleteMultiSelect.clear-button" cursor="pointer" onClick={onReset} />
+        ) : isOpen ? (
+          <IconChevronUp
+            data-testid="cmpsr.autocompleteMultiSelect.chevron-up"
+            cursor="pointer"
+            {...getToggleButtonProps()}
+          />
+        ) : (
+          <IconChevronDown
+            data-testid="cmpsr.autocompleteMultiSelect.chevron-down"
+            cursor="pointer"
+            {...getToggleButtonProps()}
+          />
+        )
```

Comment:
Could we extract this into a constant component that will return the correct component based on the isOpen?

Reply:
Is this what you had in mind? 🧐😊